code-challenges.test.js

```diff
@@ -11,40 +11,124 @@
+// We want to make sure that the function is returning the correct value.
+// Start with a named function that returns the function body.
 
-const votes1 = {upVotes: 13, downVotes: 2}
+// const votes1 = {upVotes: 13, downVotes: 2}
 // Expected output: 11
-const votes2 = {upVotes: 2, downVotes: 33}
+// const votes2 = {upVotes: 2, downVotes: 33}
 // Expected output: -31
 
 
+const votes1 = {upVotes: 13, downVotes: 2}
+const votes2 = {upVotes: 2, downVotes: 33}
+
+describe("tallyVotes", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally") => {
+    expect(tallyVotes(votes1)).toBe(11)
+    expect(tallyVotes(votes2)).toBe(-31)
+  }
+})
+
+ReferenceError: tallyVotes is not defined
+
 // b) Create the function that makes the test pass.
 
+// function combineArrays(array1, array2) {
+function tallyVotes(votes) {
+  // Create a variable that will hold the value of the upVotes and downVotes. Using the .reduce method to add the values of the upVotes and downVotes.
+  let tally = 0;
+// Using for in loop to iterate through the object and add the values of the upVotes and downVotes. i = 0 is the starting point, i < votes.length is the stopping point, and i++ is the increment.
+  for (let i = 0; i < votes.length; i++) {
+    tally += votes[i];
+  }
+// Return the tally variable.
+  return tally;
+}
+
+// Tests are passing.
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+// const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+// const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
+describe("noDuplicates", function() {
+  it("takes in two arrays as arguments and returns one array with no duplicate values", () => {
+    expect(dataArray1(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+}
+
+Reference Error: noDuplicates is not defined
 
 // b) Create the function that makes the test pass.
+
+// create a function "noDuplicates" that takes in two arrays as arguments and returns one array with no duplicate values.
+
+  // Now using a spead operator to pass in a dynamic number of arguments. 
+  const arrayPieceTogether = (array1, array2) => {
+  // Create a variable that will hold the value of the two arrays combined using the .concat method
+    (dataArray1).concat(dataArray2) 
+  // Using the build in method of spread operator to pass in a dynamic number of arguments.
+    array3 = [...new Set([...dataArray1, ...dataArray2])]
```

Comment:
The logic looks good here.  Keep in mind that you are setting a parameter of array1 and array 2 but it does not look it is ever being used in the function.  Consider how this could be refactored to be more dynamic and using the parameters set.